os/board/common/crashdump.c

```diff
@@ -53,11 +69,138 @@
  * Private Variables
  ****************************************************************************/
+#if defined(CONFIG_BOARD_RAMDUMP_FLASH)
+static char ramdump_partname[MTD_PARTNAME_LEN + 1];
+static uint32_t ramdump_partoffset = 0;
+static uint32_t ramdump_partsize = 0;
+#endif
+
+#if defined(CONFIG_BOARD_COREDUMP_FLASH)
+static char coredump_partname[MTD_PARTNAME_LEN + 1];
+static uint32_t coredump_partoffset = 0;
+static uint32_t coredump_partsize = 0;
+#endif
 
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
 
+#if defined(CONFIG_BOARD_COREDUMP_FLASH)
+static int coredump_to_flash(uint32_t cur_sp, void *tcb)
+{
+	uint32_t *ptr;
+	int32_t i;
+	uint32_t j;
+	uint32_t len = 0;
+	char pbuf[COREDUMP_BUF_SIZE];
+	int ret = -1;
+	int32_t coredump_size = 0;
+
+	sched_lock();
+
+	/* Add dummy coredump_size at the beginning */
+	len += sprintf(pbuf + len, "%08x\n", coredump_size);
+
+	/* space for '\0' character added by sprintf */
+	len += 1;
+
+	/* Process and format the stack to readable format */
+	len += sprintf(pbuf + len, "up_dumpstate:   size: %08x\n", \
+				((struct tcb_s *)tcb)->adj_stack_size);
+
+	ptr = (uint32_t *)cur_sp;
+	for (i = 0; i < 8; i++) {
+		len += sprintf(pbuf + len, "up_stackdump: %08x:", ptr);
+		for (j = 0; j < 8; j++) {
+			len += sprintf(pbuf + len, " %08x", ptr[j]);
+		}
+		len += sprintf(pbuf + len, "\n");
+		ptr += 8;
+	}
+
+	ptr = (uint32_t *)current_regs;
+	for (i = 0; i < 2; i++) {
+		len += sprintf(pbuf + len, "up_registerdump: R%d:", (i * 8));
+		for (j = 0; j < 8; j++) {
+			len += sprintf(pbuf + len, " %08x", ptr[j]);
+		}
+		len += sprintf(pbuf + len, "\n");
+		ptr += 8;
+	}
+
+	if (coredump_partsize < len) {
+		lldbg("Partition size 0x%x is less coredump size 0x%x\n", coredump_partsize, len);
+		goto err ;
+	}
+
+	/* calculate total formated coredump_size, without size field(10) bytes */
+	coredump_size = len - 10;
+
+	/* Add the actual crashlog_size at the beginning */
+	sprintf(pbuf, "%08x", coredump_size);
+
+	ret = up_dumptoflash(coredump_partoffset, (uint32_t *)pbuf, len);
+	if (ret) {
+		lldbg("Failed to write %s: len %d, errno %d\n", coredump_partname, len, errno);
+	}
+
+err:
+	sched_unlock();
+	lldbg("core dump of %d bytes dumped to partition %s\n", len, coredump_partname);
+	return ret;
+}
+#endif
+
+#if defined(CONFIG_BOARD_RAMDUMP_FLASH)
+static int ramdump_to_flash(uint32_t ram_address, uint32_t ram_size)
+{
+	uint32_t offset = 0;
+	uint32_t total_size = 0;
+	uint32_t size;
+	int ret = -1;
+
+	lldbg("ramdump size = %d bytes\n", size);
+
+	sched_lock();
+
+	if (ramdump_partsize < ram_size) {
+		lldbg("Partition size 0x%x is less ramdump size 0x%x\n", ramdump_partsize, ram_size);
+		goto err ;
+	}
+
+	/* Maximum ramsize allowed */
+	size = (ram_size < ramdump_partsize) ? ram_size : ramdump_partsize;
+
+	while (size) {
```

Comment:
please refer to previous comment regarding mtd->write.
Also inside of mtd driver, there's loop to handle bigger size buffer than size of page write.
So you don't have to add this logic, driver will handle everything, you can just throw buffer & size of buffer. 
That's all.